### DISCUSSION TITLE
Make key overlay show four keys

### DISCUSSION THREAD

**Discussion by Velyro** [ppy/osu#21179] [Ideas] [answered]
When playing osu standard with 4 keys, it would be helpful for replays if the key overlay showed all the buttons you use or at least a different colour if you use the alternate key for that button so that you can check if you were rolling properly. Sorry if this is basically the same as "let the key overlay be customisable" which I saw has already been requested.

**Comment by Doksha** [comment, marked as answer]
this is basically the same as [this](https://github.com/ppy/osu/discussions/13043) suggestion.
It would be nice if it were just 4 keys like in stable (maybe as a toggle option in settings), the merging of the keys is quite annoying for any playstyle that doesnt just use the base 2 keyboard keys (for example, tap xz, asdf roll, mouse slow key fast, and even just showing when a player alternates mouse and key, such as haga, to differentiate between different play styles)
although implementation may be difficult since the 4 key binds are 2 pairs of binds linked to 2 keys, it would have to be altered to be either 4 separate keys, or to record separate keys as actually separate. (I havent actually checked this, it might already be the case, could someone try this; set a play with 4 keys, then save the replay, and watch it in stable)

**Reply by ghost** [reply]
> could someone try this; set a play with 4 keys, then save the replay, and watch it in stable
everything is recorded as mouse buttons iirc

**Reply by Velyro** [reply]
> could someone try this; set a play with 4 keys, then save the replay, and watch it in stable)
Yeah, I checked it too, seems like when importing from lazer to stable, it just uses m1 and m2 even though it's asdf

**Reply by frenzibyte** [reply]
Marking as duplicate.
